js/modules/k6/experimental/tracing/client_test.go

```diff
@@ -0,0 +1,166 @@
+package tracing
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/js/modulestest"
+)
+
+// traceParentHeaderName is the normalized trace header name.
+// Although the traceparent header is case insensitive, the
+// Go http.Header sets it capitalized.
+const traceparentHeaderName string = "Traceparent"
+
+// testTraceID is a valid trace ID used in tests.
+const testTraceID string = "00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-00"
+
+func TestClientInstrumentArguments(t *testing.T) {
+	t.Parallel()
+
+	t.Run("no args should fail", func(t *testing.T) {
+		t.Parallel()
+
+		testCase := newTestCase(t)
+
+		_, err := testCase.client.instrumentArguments(testCase.traceContextHeader)
+		require.Error(t, err)
+	})
+
+	t.Run("1 arg should initialize params successfully", func(t *testing.T) {
+		t.Parallel()
+
+		testCase := newTestCase(t)
+		rt := testCase.testSetup.VU.Runtime()
+
+		gotArgs, gotErr := testCase.client.instrumentArguments(testCase.traceContextHeader, goja.Null())
+
+		assert.NoError(t, gotErr)
+		assert.Len(t, gotArgs, 2)
+		assert.Equal(t, goja.Null(), gotArgs[0])
+
+		gotParams := gotArgs[1].ToObject(rt)
+		assert.NotNil(t, gotParams)
+		gotHeaders := gotParams.Get("headers").ToObject(rt)
+		assert.NotNil(t, gotHeaders)
+		gotTraceParent := gotHeaders.Get(traceparentHeaderName)
+		assert.NotNil(t, gotTraceParent)
+		assert.Equal(t, testTraceID, gotTraceParent.String())
+	})
+
+	t.Run("2 args with null params should initialize it", func(t *testing.T) {
+		t.Parallel()
+
+		testCase := newTestCase(t)
+		rt := testCase.testSetup.VU.Runtime()
+
+		gotArgs, gotErr := testCase.client.instrumentArguments(testCase.traceContextHeader, goja.Null(), goja.Null())
+
+		assert.NoError(t, gotErr)
+		assert.Len(t, gotArgs, 2)
+		assert.Equal(t, goja.Null(), gotArgs[0])
+
+		gotParams := gotArgs[1].ToObject(rt)
+		assert.NotNil(t, gotParams)
+		gotHeaders := gotParams.Get("headers").ToObject(rt)
+		assert.NotNil(t, gotHeaders)
+		gotTraceParent := gotHeaders.Get(traceparentHeaderName)
+		assert.NotNil(t, gotTraceParent)
+		assert.Equal(t, testTraceID, gotTraceParent.String())
```

Comment:
nit: In general I prefer to user `require` as it will stop the execution directly which helps in two ways:
1. I won't get a wall of text when i break somethign listing 20 things that went wrong, when the first thing that went wrong is likely to tell me what I neeed to know to fix it.
2. In this particular case `assert.NotNil` failing means that hte test *will* panic with nil exception. Arguably you can mix assert and require, but I have rarely been in a case taht this makes sense :shrug:

Reply:
In general, I try to use `require` when it's a requirement for the test, as in: "this is required even to perform the test in the first place", and use `assert` for stuff that's tested. I guess I approach it in a rather semantic way, "this is required to perform the test", and "this asserts the behavior of the tested code is correct".

It's also how I understood it to be used in the tests I've been exposed.

Do you have a  strong opinion on this, or would it be fine if I keep it like this? I'm okay with either way.